### PR TITLE
Swap module simulations: swap msg operations

### PR DESCRIPTION
**Pull request by denalimarsh**
This PR contains
1. Simulated `MsgSwapExactForTokens` operation
2. Simulated `MsgSwapForExactTokens` operation
3. Swap simulation decoder + test

### CODE REVIEW

File: x/swap/simulation/operations.go [hunk 1]
```diff
@@ -258,6 +266,204 @@ func SimulateMsgWithdraw(ak types.AccountKeeper, k keeper.Keeper) simulation.Ope
...
+	return func(
+		r *rand.Rand, app *baseapp.BaseApp, ctx sdk.Context, accs []simulation.Account, chainID string,
+	) (simulation.OperationMsg, []simulation.FutureOperation, error) {
+
+		poolRecords := k.GetAllPools(ctx)
+		r.Shuffle(len(poolRecords), func(i, j int) {
+			poolRecords[i], poolRecords[j] = poolRecords[j], poolRecords[i]
+		})
+
+		// Find an account-pool pair for which withdraw is possible
```
Comment (karzak) [inline x/swap/simulation/operations.go:379]: same as above

**Review by karzak — APPROVED** [review]
tACK

File: x/swap/simulation/operations.go (hunk 1 above)
Comment (denalimarsh) [reply x/swap/simulation/operations.go]: https://github.com/Kava-Labs/kava/pull/972/commits/b5f6fe0fa9063f18dfc3eef0b77f9b83c4328a21

File: x/swap/simulation/operations.go [hunk 2]
```diff
@@ -258,6 +266,204 @@ func SimulateMsgWithdraw(ak types.AccountKeeper, k keeper.Keeper) simulation.Ope
...
```
[the remark this answers is not in the document]
Comment (denalimarsh) [reply x/swap/simulation/operations.go]: https://github.com/Kava-Labs/kava/pull/972/commits/b5f6fe0fa9063f18dfc3eef0b77f9b83c4328a21

**Pull request merged**